config/confighttp/confighttp.go

```diff
@@ -131,6 +130,8 @@ func (hss *HTTPServerSettings) ToServer(handler http.Handler) *http.Server {
 		co := cors.Options{AllowedOrigins: hss.CorsOrigins}
 		handler = cors.New(co).Handler(handler)
 	}
+	// TODO: Make this configurable via HTTPServerSettings.
+	handler = HttpBodyDecompressor(handler)
```

Comment:
I'm not sure how we can make this configurable via the settings. Would something like this work?
_defaults for OTLP receiver_:
```
http:
   decompression:
      enable: true
      error_format: otlp
```
_other receivers_ if they want to use the decompression middleware:
```
http:
   decompression:
      enable: true
```

Reply:
I don't think we need to make this configurable. OTLP spec is pretty clear about this behavior, there is no need to make it configurable.

Reply:
Thanks for a lot for clarifying that!

I had a question on handling error response for OTLP vs other receivers(like zipkin) as they differ in how they respond with a 400 bad request error. 

Is having a  `ToServerOption` in `HTTPServerSettings.ToServer()` to configure the `errorHandler` for `HttpBodyDecompressor` the right approach?  I could not find a way to infer the receiver type from the headers which would have been convenient. 

With this, we can have the OTLP specific error handler in the module where the receiver is implemented.

Reply:
>Is having a ToServerOption in HTTPServerSettings.ToServer() to configure the errorHandler for HttpBodyDecompressor the right approach? I

Yes, I think it is a reasonable approach.

>With this, we can have the OTLP specific error handler in the module where the receiver is implemented.

I think this is highly desirable.

Reply:
Sorry for the delay. I've updated the PR factoring in the suggestions made. Hope it looks better now.